packages/block-library/src/link-preview/edit.js

```diff
@@ -0,0 +1,128 @@
+/**
+ * WordPress dependencies
+ */
+import { __, _x } from '@wordpress/i18n';
+import { useState, useEffect } from '@wordpress/element';
+import { useBlockProps, BlockControls } from '@wordpress/block-editor';
+import { __experimentalFetchUrlData } from '@wordpress/core-data';
+import {
+	Placeholder,
+	Spinner,
+	Button,
+	ToolbarGroup,
+	ToolbarButton,
+} from '@wordpress/components';
+import { link, edit } from '@wordpress/icons';
+
+export default function LinkPreviewEdit( props ) {
+	const {
+		attributes: { url, title, icon, image },
+		setAttributes,
+	} = props;
+	const [ isFetching, setIsFetching ] = useState( false );
+	const [ isEditingUrl, setIsEditingUrl ] = useState( ! url );
+	const [ urlValue, setURLValue ] = useState( '' );
+
+	const blockProps = useBlockProps( {
+		href: url,
+		className: image ? 'has-image' : undefined,
+		onClick: isEditingUrl
+			? undefined
+			: ( event ) => {
+					event.preventDefault();
+			  },
+	} );
+
+	useEffect( () => {
+		if ( url && ! title && ! icon && ! image ) {
+			setIsFetching( true );
+			__experimentalFetchUrlData( url )
+				.then( ( data ) => {
+					setAttributes( data );
+				} )
+				.finally( () => {
+					setIsFetching( false );
+				} );
+		}
+	}, [] );
```

Comment:
It seems this will only ever run once because there are no dependencies listed. If the `url` changes don't we want to re-fetch the `data` with using the new URL? 
```suggestion
	}, [ url, title, icon, image ] );
```

Reply:
Removed the effect, see below.